bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -3360,6 +3566,8 @@ private static boolean checkJSONCast(BValue json, BType sourceType, BType target
             case TypeTags.FLOAT_TAG:
             case TypeTags.BOOLEAN_TAG:
                 return json != null && json.getType().getTag() == targetType.getTag();
+            case TypeTags.DECIMAL_TAG:
```

Comment:
I think we may need to clarify this, like how do we keep decimals in json, IMO we should keep them as decimal, then both decimal and float will work. But if we keep it as float, only float will work properly, and when it comes to decimal, precision will be lost.